go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/fsnotify/fsnotify v1.5.1
 	github.com/go-logr/logr v1.2.0
 	github.com/go-logr/zapr v1.2.0
-	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/ginkgo/v2 v2.1.1
```

Comment:
Is not it a breaking change?
Will not this change affects who is using C+R and testing their projects with ENV TEST for example?
Will not be required to migrate the code to do the same with v2 as well?

Reply:
No. The package is only used in tests. All changed files are also `_test.go` files and those can't be imported by other packages/programs. So there is no way this would affect consumers of controller-runtime.

The dependency in `go.mod` also doesn't seem to have an effect, at least cluster-api tests are still executing fine when using a `replace` directive.

Reply:
@schrej,

Controller runtime provides EnvTest, which allows us to write the tests.
It seems that it is a breaking change ( changes the behaviour for who is importing kubebuilder and writing tests with. Am I right? if yes, I think we need to clarify that in the title as ⚠️ at least.  )

Reply:
Ah, you are right. There actually is one package that does import ginkgo outside of tests: `pkg/envtest/printer`
But it seems like the `v2` suffix of the new version allows to import both versions at the same time. At least the compiler didn't complain when I used this PR with cluster-api, and that one is still on v1.

Reply:
Note that the ginkgo made a MAJOR bump. That is only required when we have breaking changes ( unless you have a version that is 0.MINOR releases which means initial development )

The MAJOR bump occurred because they removed some deprecated options.

Therefore, consumers of controller runtime might be impacted because they will need to update and change their tests. So, we need to update the title of this PR ( which is used for generating the release notes ) with ⚠️ instead of use ✨. Also, it would be nice to have in the description of the PR the link for the release notes of the dep as the guidance to upgrade from v1 to v2.